spec/duck-model/duck_spec.rb

```diff
@@ -23,6 +23,38 @@
 
 end
 
+describe Duck do
+  describe "when re-arranging ranks " do
+    context "cannot be re-arranged" do
+
+      it "fails to save" do
+        patched_duck = Duck.new(
+          :name => 'Quacky',
+          :pond => 'Shin',
+          :age_position => 1,
+        )
+        age_ranker = Duck.rankers.detect {|ranker| ranker.name == :age }
+        Duck.expects(:rankers).once.returns([age_ranker])
+        patched_duck_ranker = age_ranker.with(patched_duck)
+        age_ranker.expects(:with).once.returns(patched_duck_ranker)
+        class << patched_duck_ranker
+          def current_first
+            nil
+          end
+
+          def current_at_rank(_rank)
+            true # something truthy
```

Comment:
This is otherwise nil. It's need to be truthy for `rearrange_ranks` to be called